packages/smart-contracts/test/contracts/ERC20SwapToPay.test.ts

```diff
@@ -118,6 +120,47 @@ describe('contract: SwapToPay', () => {
     expect(finalContractPaymentBalance.toNumber()).to.equals(0);
     expect(finalContractRequestBalance.toNumber()).to.equals(0);
   });
+  it('swaps and pays the request and swapfee', async function () {
+    const initialFeeBalance = await spentErc20.balanceOf(swapFeeAddress);
+
+    await expect(
+      testSwapToPay.swapTransferWithReference(
+        to,
+        100,
+        // Here we spend 206 max, for 202 used in theory, to test that 4 is given back
+        206,
+        [spentErc20.address, paymentNetworkErc20.address],
+        referenceExample,
+        1,
+        builder,
+        exchangeRateOrigin + 1000, // _uniswapDeadline. 100 -> 1000: Too low value may lead to error (network dependent)
+      ),
+    )
+      .to.emit(erc20FeeProxy, 'TransferWithReferenceAndFee')
+      .withArgs(
+        ethers.utils.getAddress(paymentNetworkErc20.address),
+        to,
+        '100',
+        ethers.utils.keccak256(referenceExample),
+        '1',
```

Comment:
```suggestion
        '2',
```
I think it's wise to test with different expected amounts to be sure we never mix both kinds of fees.